### PR TITLE
Full Template Wavelength Calibration

### PR DESCRIPTION
**Pull request by profxj**
Introduces a new method which is a riff on `reidentify` to calibrate
multi-slit spectra (and longslit).  It is called `full_template` and is 
now the default mode for `shane_kast_blue`,
`keck_lris_blue`, `keck_deimos` and a few of the `keck_lris_red` gratings.

Here are the specific changes:

- Introduce `full_template()` method for multi-slit wavelength calibrations; includes `nsnippet` parameter
- Generate full template files for LRIS, DEIMOS, Kastb
- Added a few new Arc lines for DEIMOS in the blue

New docs too.  No new tests. 
This should have no effect on instruments other than the 
ones listed above except for the new criterion on good arclines.

This PR is coupled with one in the Dev Suite (the `lrisb_test_reid` branch).

Addresses Issues #506 and #507

### CODE REVIEW

**Comment by jhennawi** [comment]
Looks great X. Thanks for the hard work here. Can you please ensure that the *entire* dev suite script runs through before merging. We need to be sure that the minor modification you made to the arc line identification does not impact other setups.

File: doc/wave_calib.rst [hunk 1]
```diff
@@ -122,68 +157,51 @@ More details are provided in :doc:`heliocorr`.
...
+Instrument       Setup                      Name
+===============  =========================  =============================
+keck_deimos      600ZD grating, all lamps   keck_deimos_600.fits
+keck_deimos      830G grating, all lamps    keck_deimos_830G.fits
+keck_deimos      1200G grating, all lamps   keck_deimos_1200G.fits
+keck_lris_blue   B300 grism, all lamps      keck_lris_blue_300_d680.fits
+keck_lris_blue   B400 grism, all lamps?     keck_lris_blue_400_d560.fits
+keck_lris_blue   B600 grism, all lamps      keck_lris_blue_600_d560.fits
+keck_lris_blue   B1200 grism, all lamps     keck_lris_blue_1200_d460.fits
+keck_lris_red    R400 grating, all lamps    keck_lris_red_400.fits
```
[the remark this answers is not in the document]
Comment (profxj) [reply doc/wave_calib.rst]: That is right.

**Comment by profxj** [comment]
Can't run the full suite from Australia.
Can you do so on your machinie @jhennawi ?
Thanks.

File: pypeit/wavecalib.py [hunk 2]
```diff
@@ -206,7 +210,13 @@ def build_wv_calib(self, arccen, method, skip_QA=False):
             # Now preferred
             arcfitter = wavecal.autoid.ArchiveReid(arccen, par=self.par, ok_mask=ok_mask)
             patt_dict, final_fit = arcfitter.get_results()
-
+        elif method == 'full_template':
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit/wavecalib.py]: I have added to the docs `wave_calib.rst`
check that out.

**Pull request merged**